### DISCUSSION TITLE
[Suggestion] Add possiblity for a SSO user to have his email updated

### DISCUSSION THREAD

**Discussion by Maypul** [firefly-iii/firefly-iii#8740] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)
- [X] My request is not listed as [a very good idea, but unfortunately...](https://docs.firefly-iii.org/explanation/more-information/what-its-not/)
- [X] I've used [the search](https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue) and this has not been requested before.

### Description

I would like Firefly III to either update user's email account automatically or allow to update it manually if the user is a SSO user.

Currently, Firefly III seems to create user using provided email address from the header and even if the email address is updated in the SSO provider, it is not being updated in this app. And since it is a SSO user, option to update in UI is not available.

### Solution

AUTHENTICATION_GUARD_EMAIL: HTTP_X_AUTHENTIK_EMAIL - if information in this header is changed, also update email address of the user in the app.

### What are alternatives?

Most likely update it manually in the database. But when I had it set up this way:
AUTHENTICATION_GUARD_HEADER: HTTP_X_AUTHENTIK_USERNAME
AUTHENTICATION_GUARD_EMAIL: HTTP_X_AUTHENTIK_EMAIL
in the database in the user's fields there was an username, so I would guess the email is also stored somewhere else as it was shown correctly in the admin panel.

**Comment by JC5** [comment, marked as answer]
This will always require manual database intervention. I can put something in the FAQ, if you like?

The reason is simple:

> if information in this header is changed, also update email address of the user in the app.

Firefly III doesn't know the information in this header is changed, because it **cannot see** the difference between the previous request and the current request: that's the job of Authentik. 

There is nothing connecting you to the old information when you show up with new information.

**Comment by Maypul** [comment]
Thanks. Would it be maybe feasible to add additional identification to the user table so we could identify users via username instead since we have already AUTHENTICATION_GUARD_HEADER / AUTHENTICATION_GUARD_EMAIL so in this case email would not be used for identification and could be enabled to be changed via application for the SSO user?
